src/plugins/dataflow/nodes/utilities/live-output-utilities.ts

```diff
@@ -12,11 +17,52 @@ export function getHubSelect(node: Node) {
 
 export function getOutputType(node: Node) {
   const outputTypeControl = node.controls.get("liveOutputType") as DropdownListControl;
-  return outputTypeControl.getValue();
+  if (outputTypeControl) return outputTypeControl.getValue();
 }
 
 export function getNodeValueWithType(node: Node): NodeOutputValue {
   const val = node.data.nodeValue as number;
   const outType = getOutputType(node);
   return { val, outType };
 }
+
+export function outputsToAnyRelay(node: Node) {
+  return kMicroBitHubRelaysIndexed.includes(getOutputType(node));
+}
+
+export function outputsToAnyGripper(node: Node) {
+  return kGripperOutputTypes.includes(getOutputType(node));
+}
+
+export function getLiveOptions(node: Node, sharedVar?: VariableType, device?: string | null) {
+  const options: any[] = [];
+  const simOption = sharedVar && simulatedHub(sharedVar);
+  const anyOuputFound = simOption || device === "arduino" || device === "microbit";
+  const { liveGripperOption, warningOption } = baseLiveOutputOptions;
+
+  if (sharedVar) {
+    options.push(simOption);
+  }
+
+  if (outputsToAnyRelay(node) && device === "microbit") {
+    options.push(...NodeMicroBitHubs);
+  }
+
+  if (outputsToAnyGripper(node) && device === "arduino") {
+    options.push(liveGripperOption);
+  }
+
+  if (!anyOuputFound) options.push(warningOption);
+
+  return options;
+}
+
+export function setLiveOutputOpts(node: Node, device: string | null, sharedVar?: VariableType) {
+  const hubSelect = getHubSelect(node);
+  const options = getLiveOptions(node, sharedVar, device);
+  const selectedOption = options.find(option => option && option.name === hubSelect.getValue());
+  const firstOption = !selectedOption ? options[0] : undefined;
+  if (firstOption) hubSelect.setValue(firstOption.name);
```

Comment:
```suggestion
  if (!selectedOption) hubSelect.setValue(options[0]?.name);
```
Note sure if you need that `?` but seems like the safe move.

Reply:
I kept it as it was in the end - because otherwise we can end up with no value in the option, which causes the hubSelect to disappear, which felt wonkier than having it show with the only potential option greyed out.